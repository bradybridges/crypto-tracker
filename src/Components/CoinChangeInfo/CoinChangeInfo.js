import React from 'react';
import PropTypes from 'prop-types';
import './CoinChangeInfo.scss';
import upArrow from '../../Images/up-arrow.png';
import downArrow from '../../Images/down-arrow.png';

const CoinChangeInfo = ({ data, timeFrame }) => {
  const priceImg = data.price_change > 0 ? upArrow : downArrow;
  const volumeImg = data.volume_change > 0 ? upArrow : downArrow;
  const marketImg = data.market_cap_change > 0 ? upArrow : downArrow;
  const priceChange = Number(data.price_change).toFixed(2);
  const priceChangePct = (Number(data.price_change_pct) * 100).toFixed(2);
  const volChange = Number(data.volume_change).toFixed(2);
  const volChangePct = (Number(data.volume_change_pct) * 100).toFixed(2);
  const marketChange = Number(data.market_cap_change).toFixed(2);
  const marketChangePct = (Number(data.market_cap_change_pct) * 100).toFixed(2);

  return (
    <section id="coin-change-info-container">
      <h2>
        Change in
        {timeFrame}
      </h2>
      <div className="info-container">
        <p>Price</p>
        <p>
          $
          {priceChange}
        </p>
        <p>
          %
          {priceChangePct}
        </p>
        <img src={priceImg} alt="price indicator" />
      </div>
      <div className="info-container">
        <p>Volume</p>
        <p>{volChange}</p>
        <p>
          %
          {volChangePct}
        </p>
        <img src={volumeImg} alt="volume indicator" />
      </div>
      <div className="info-container">
        <p>Market Cap</p>
        <p>
          $
          {marketChange}
        </p>
        <p>
          $
          {marketChangePct}
        </p>
        <img src={marketImg} alt="market indicator" />
      </div>
    </section>
  );
}

export default CoinChangeInfo;

CoinChangeInfo.propTypes = {
  data: PropTypes.object.isRequired,
  timeFrame: PropTypes.string.isRequired,
};
