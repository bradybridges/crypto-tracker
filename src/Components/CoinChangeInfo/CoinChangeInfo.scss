#coin-change-info-container {
  width: 90%;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 10px;
}

.info-container {
  width: 100%;
  display: flex;
  align-items: center;
  background-color: #3e3e3ed4;
}

.indicator {
  margin-right: 5px;
}