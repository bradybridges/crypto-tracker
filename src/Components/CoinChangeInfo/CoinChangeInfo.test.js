import React from 'react';
import { shallow } from 'enzyme';
import CoinChangeInfo from './CoinChangeInfo';

describe('CoinChangeInfo', () => {
  let wrapper;
  const mockData = {
    price_change: 300.05,
    price_change_pct: 4.23,
    volume_change: 10000,
    volume_change_pct: 10.3,
    market_cap_change: -425232,
    market_cap_change_pct: -1.1,
  };
  beforeEach(() => {
    wrapper = shallow(<CoinChangeInfo data={mockData} timeFrame="1d" />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
