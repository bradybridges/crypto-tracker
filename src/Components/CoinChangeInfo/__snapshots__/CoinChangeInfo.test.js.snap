// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinChangeInfo should match the snapshot 1`] = `ShallowWrapper {}`;
